scripts/zulip_emoji_merge_delegate.py

```diff
@@ -50,7 +50,25 @@
 print(f"public_response:{public_response}")
 print(f"reviewers_response:{reviewers_response}")
 
-messages = (public_response['messages']) + (reviewers_response['messages'])
+# Get the oldest message in any topic in `PR reviews` whose title matches `#{PR_NUMBER}`.
+first_message_in_PR_reviews = client.get_messages({
+    "anchor": "oldest",
+    "num_before": 0,
+    "num_after": 1,
+    "narrow": [
+        {"operator": "channel", "operand": "PR reviews"},
+        {"operator": "search", "operand": f'#{PR_NUMBER}'},
+    ],
+})
+print(f'First message:{first_message_in_PR_reviews}')
+
+messages = (public_response['messages']) + (reviewers_response['messages']) + (first_message_in_PR_reviews['messages'])
+
+print(messages)
+# we use the ID of the first message in the `PR reviews` channel to react to it
+# whether or not it has the PR number in the content.
+first_message_in_PR_reviews_id=messages[0]['id']
```

Comment:
Why only the first message?

Reply:
I think that the idea was that the bot would react *somewhere* on the thread, plus wherever the PR number is mentioned explicitly.

If you think that reacting more is useful, that is easy to change!

Reply:
@eric-wieser I had not understood your comment: you had indeed identified a bug!

Only the first message *in the relevant `PR review` thread* gets a reaction, even if it does not mention the PR number.  The rest of the emoji reactions are still applied as normal!